### ISSUE TITLE
Replicate Java primitive-wrapper classes such as Integer

### ISSUE THREAD

**Issue by teo1978** [processing-js/processing-js#180]
Sometimes one needs to do things as simple as this:
```
String binaryString="001110010";
int i=Integer.parseInt(binaryString,2);
```

That works in Processing but gives unexpected results in ProcessingJS.

Note that Processing's own parseInt (i.e. PApplet.parseInt(String, int)) doesn't work like Integer.parseInt(String, int). That is sad and pretty questionable on Processing's side, but change that would break bc, I guess.

I know it's probably a lot of work, but using primitive type wrapper classes is almost inevitable in Java, and hence in Processing. Even though Processing JS can't possibly replicate every class in the standard Java libraries (or whatever they are called), I think wrapper classes of primitive types (Integer, Boolean, etc) are a must. I can hardly think of a sketch of a serious complexity that won't break in PJS because of this.

**Comment by Pomax** [comment]
this is actually something that you can shim instantly by just adding your own script:
```
<script>
var Integer = {
  parseInt: function(v,radix) { return window.parseInt(v,radix); }
};
</script>
```
I'd recommend this as first approach when using out-of-processing calls to the Java subsystem.

**Comment by teo1978** [comment]
Nope.
That's a good workaround, but you shouldn't dismiss this issue.
For non-expert users, this is not a solution. They wouldn't even know where the issue comes from.
They would just see yet another simple sketch that works perfectly in Processing and breaks when exported via PJS.

This solution is valid for the big world of Java libraries, where one knows that he's writing code that depends on Java-specific stuff. But primitive-type wrapper classes are something so basic and vital that they shouldn't  even be considered  as "out-of-processing".

**Comment by Pomax** [comment]
This issue has come up in various forms in the past, and the Pjs philosophy is still the same: if you know enough about Java to reach into Java's (not Processing's) objects, then now the burden lies on you. If someone wants to write shims to be included with Pjs, great, but Pjs is not going to offer the implementations for not-Processing objects out of the box.

**Comment by teo1978** [comment]
Yeah, I get that for things such as HashSet, but you should seriously reconsider for the primitive wrappers.

I don't "think Processing and Java are the same", I know the difference pretty well. But the limit where Processing's "programming model" ends and "pure java" starts can be somewhat blurry. Since Processing translates into java, a lot of things come "for free" in Processing just because they are as they are in java. What is and is not in the Processing reference is in many cases just a matter of how much rigur Processing authors wanted to put in the effort of writing the documentation.

I'll make an example.
HashMap is in the Processing reference. The authors of Processing didn't write it, it's part of Java's standard library. They just deemed it important enough to mention it in the reference.
OTOH, they didn't deem HashSet (or any other implementation of Set) as important, or more probably, they forgot about it. I seriously dubt they put much effort in deciding whether or not HashSet was as important as HashMap, and that they decided it wasn't based on a thorough analysis.
As a result, for you HashMap is part of the Processing API, while HashSet is not. Hence you implement HashMap in Processing.js but not HashSet.

Your criterion, "if it's in the Processing reference then it's part of Processing", is admittedly the only easy and objective one, but by taking the Processing reference as the Bible, you're granting too much authority to those who wrote the reference. The fact that one useful class is in the Processing reference and another is not, is in many cases not because one was thoughtfully decided to be "part of the Processing programming model" and the other wasn't, it's just that one day they thought about including it in the reference, while they overlooked the other.

The bottom line is that some classes are not in the reference today, but may be tomorrow. I understand your approach is "well, I'll add them to PJS tomorrow when they add them to the Processing API". But when "adding to the processing API" means nothing but "adding them to the Reference", it's questionable to say that "they don't belong to the Processing API". It may just be that nobody has taken care to write a page for them in the reference.

And it is the case, in my opinion, of primitive-wrapper classes.

**Comment by Pomax** [comment]
noted, and will leave this open for someone to take, or for me to take when everything else has been done.
